openedx_learning/apps/authoring/publishing/api.py

```diff
@@ -493,3 +495,29 @@ def ready(self):
     return PublishableContentModelRegistry.register(
         content_model_cls, content_version_model_cls
     )
+
+
+def get_drafts(publishable_entities: QuerySet[PublishableEntity]) -> QuerySet[Draft]:
+    """
+    Given a list of publishable entities returns current drafts.
+    Not all entities may have current drafts.
+    """
+    entity_ids = publishable_entities.values_list("id", flat=True)
+
+    return Draft.objects.filter(
+        entity_id__in=entity_ids,
+        version__isnull=False,
+    ).select_related("entity", "version")
```

Comment:
I was thinking about this more recently. The `entity_id__in` query is reasonable for what this does, but it does get expensive for larger queries. But I think even more importantly, the fact that we'd be passing back a `QuerySet[Draft]` makes it harder for callers to chain queries.

Maybe something like this instead?

```python
def filter_entities(entities: QuerySet[PublishableEntity], has_draft=None, has_published=None) -> QuerySet[PublishableEntity]:
    if has_draft is not None:
        entities = entities.filter(draft__version__isnull=(not has_draft))
    if has_published is not None:
        entities = entities.filter(published__version__isnull=(not has_published))

    return entities
```

Reply:
With the calling code from https://github.com/openedx/edx-platform/pull/35734/files looking something like:

```python
        draft_num_children = authoring_api.filter_entities(collection.entities, has_draft=True).count()
        published_num_children = authoring_api.filter_entities(collection.entities, has_published=True).count()
```

Reply:
Thanks for the suggestion. Updated here: https://github.com/openedx/openedx-learning/pull/257/commits/19f542b6b56b8b7e62827acaecdbf9d919f551f9